hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmapp/RMAppRunningOnNodeEvent.java

```diff
@@ -23,13 +23,23 @@
 
 public class RMAppRunningOnNodeEvent extends RMAppEvent {
   private final NodeId node;
+  private final boolean inAcquiredState;
```

Comment:
Nit: since the key here is that this container is transitioning from ACQUIRED directly to RELEASED I think Adam's fromAcquired variable naming was more descriptive.

Reply:
My impression is that in the current implementation, the inAcquiredState = true is set when the container is transitioning into the ACQUIRED state, and in this context, the flag meaning is that the container is in the ACQUIRED state. E.g. RMContainerImpl:106 -> ALLOCATED to ACQUIRED triggers the AcquiredTransition, which sets the flag to true.
If we trigger a transition event when get from the ACQUIRED state to the RELEASED state, then yes I would agree.

Reply:
You're right, I misunderstood it based on the description of the ticket, and missed that we actually call this from AcquiredTransition. 

However I still think the name is misleading, as RMAppRunningOnNodeEvent is an event, it can't be in Acquired state, this field is basically telling us if the event was created from the Acquired state transition (which I guess was Adam's intention with the first naming). So I think the name of containerInAcquiredState or createdFromAcquiredState/fromAcquiredState would be better suited.

Reply:
Got it! I like the containerInAcquiredState or createdFromAcquiredState names, it makes more sense for me this way.

Reply:
The only thing is I'm not sure if we should introduce the container concept here, as it's mostly irrelevant to the event itself (and its other uses). Hence I think the latter is a bit more self-explanatory.